### PR TITLE
Fixing expressions

### PR DESCRIPTION
**Pull request by JamesDoingStuff**
Rework the grammar for expressions to fix previous issues where an expression such as `a - b` would be parsed as the implicit multiplication `a (-b)`. This occurred because implicit multiplication and unary operators both had a greater precedence that subtraction.
Adjusting any of the precedences would result in failures with the order of operations, so the fix is to restrict the inputs of some of the expressions. This requires splitting exponentiation from the other binary operations so that it can be specified as the only binary operation permitted as an argument to the unary operator. 

Additionally:
-  The test library is expanded and reorganised, using multiple test headers per file and the `:error` tag for tests that are expected to fail
- The '+' unary operator is added

### CODE REVIEW

File: tree-sitter-topas/test/corpus/macros.inp
Comment (garryod) [inline tree-sitter-topas/test/corpus/macros.inp:1]: This too

File: tree-sitter-topas/test/corpus/numbers.inp
Comment (garryod) [inline tree-sitter-topas/test/corpus/numbers.inp:1]: And this

File: tree-sitter-topas/test/corpus/strings.inp [hunk 1]
```diff
@@ -2,12 +2,32 @@
...
+      (string_literal))
+
+======================
+Missing closing quotes
+:error
+======================
+
+"open string
+
+----------------------
```
Comment (garryod) [inline tree-sitter-topas/test/corpus/strings.inp:33]: Missing newline at bottom :stuck_out_tongue:

File: tree-sitter-topas/test/corpus/strings.inp
Comment (garryod) [inline tree-sitter-topas/test/corpus/strings.inp:1]: This too

**Review by garryod — CHANGES_REQUESTED** [review]
Looking good, would be nice to move the changes to {comment, macro, number, string} tests to another PR though

**Review by garryod — APPROVED** [review]
LGTM

File: tree-sitter-topas/test/corpus/strings.inp (hunk 1 above)
Comment (JamesDoingStuff) [reply tree-sitter-topas/test/corpus/strings.inp]: Darn, I was doing so well! :tired_face:

File: tree-sitter-topas/grammar.js [hunk 2]
```diff
@@ -80,47 +80,75 @@ module.exports = grammar({
...
+      field('operator', '^'),
+      field('right', $._expression),
+    )),
+
+    binary_expression: $ => choice(
 
+      $._exponentiation,
 
-      return choice(...table.map(({precedence, operator}) => prec.left(precedence, seq(
+      prec.left(PRECEDENCE.multiplicative, seq(
```
[the remark this answers is not in the document]
Comment (JamesDoingStuff) [reply tree-sitter-topas/grammar.js:108]: Fortunately that's something the tech spec actually documents well:
`x^y^z = (x^y)^z`

**Pull request merged**